TTWWW-473: Prevalence Map "mega dot" behavior

mattga11ant opened the pull request:
Jira ticket for reference: https://simonsfoundation.atlassian.net/browse/TTWWW-473
Figma for reference: https://www.figma.com/design/wcJ1rzoLOkRaVuKp2Zae6F/Prevalence-Map?node-id=1013-11698&node-type=frame&t=Iq4mJXmQmtiojAp1-0

- [x] pull this branch
- [x] compile CSS and JS updates
- [x] confirm that on initial load, mega dot clusters form based on dot proximity, and the dot style matches the design
- [x] confirm that the minimum mega dot cluster size is 5
- [x] hover a mega dot and confirm the style is black with white text like the design
- [x] click a mega dot to open it and confirm that it receives focus by adding 60% opacity to all other dots and mega dots
- [x] click a mega dot to open it and confirm that the mega dot outline matches the Figma design
- [x] click a mega dot to open it and confirm it contains the correct amount of dots inside it
- [x] click a mega dot to open it, the click a dot inside of the mega dot. confirm that the information panel populates with that dots info and the mega dot stays open
- [x] click a mega dot to open it, then click anywhere else on the map and confirm the mega dot closes
- [x] click a mega dot to open it, then click any other dot outside of that mega dot. confirm the mega dot closes and the info card for that other dot is populated
- [x] zoom in on a mega dot and confirm that it recalculates the mega dot size (a dot may start at 7, then you zoom in it becomes 5 and two separate dots, then more zoom it may go away when the cluster is less than 5)
- [x] click a mega dot to open it and zoom in, confirm that it closes if the mega dot size is recalculated
- [x] click a mega dot to open it, and click the clear button, confirm that the mega dot closes
- [x] click a mega dot to open it, and click anywhere inside that mega dot other than the dots it contains, and confirm the mega dot closes

mattga11ant commented:
@yjfnyc I have this mega dot functionality updated based on our conversation last week. Fixed the issue where clicking a mega dot would create a series of smaller ones. Also increased the radius a bit to make the clusters appear better, and reduced the scaling of the mega dots so that the larger ones are not quite as large. Also fixed a z-index issue with expanded mega dots so they are always on top of surrounding dots when open.

yjfnyc reviewed (changes requested):
I am still in review progress, but would like to let you know first that one testing step is not working as described. And I also have some code comments, please see if they make sense. Thanks!

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/tailwind.config.js:
```diff
@@ -320,6 +320,9 @@ module.exports = {
             },
+            fillOpacity: {
+                '15': '0.15',
+            },
```
This was left over from trying something. I removed it.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -17,6 +19,253 @@ export function ttInitMap() {
+        };
+
+        // function to zoom to a specific area
+        function zoomToArea(bbox) {
```
No longer being used. Removed.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -17,6 +19,252 @@ export function ttInitMap() {
+
+            app.map.clusters = clusters;
+
+            if (previousExpandedClusterId) {
```
Comment added.

mattga11ant commented:
> I am still in review progress, but would like to let you know first that one testing step is not working as described. And I also have some code comments, please see if they make sense. Thanks!

I have addressed all of your comments so far. Additionally fixed the clear button issue. Clicking clear now closes mega dots.

yjfnyc commented:
> > I am still in review progress, but would like to let you know first that one testing step is not working as described. And I also have some code comments, please see if they make sense. Thanks!
> 
> @yjfnyc I have addressed all of your comments so far. Additionally fixed the clear button issue. Clicking clear now closes mega dots.

I can see there are still 2 comments you haven't responded. Could you take a look when you've got a chance? Thanks!

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -17,6 +19,252 @@ export function ttInitMap() {
+                    if (nodeId === otherNodeId || visitedNodes.has(otherNodeId)) return;
+
+                    const dx = node.originalX - otherNode.originalX;
+                    const dy = node.originalY - otherNode.originalY;
```
Weird, I commented on this but may not have clicked the comment button. I tried this, but it produces a larger cluster that includes more of the outside dots when expanded. Screenshot showing this what this looks like. I did not make any changes.

<img width="84" alt="Screenshot 2025-05-02 at 3 04 06 PM" src="https://github.com/user-attachments/assets/7f41fde1-8c7d-41b9-bc12-f49a79c3f764" />

mattga11ant commented:
> > > I am still in review progress, but would like to let you know first that one testing step is not working as described. And I also have some code comments, please see if they make sense. Thanks!
> > 
> > 
> > @yjfnyc I have addressed all of your comments so far. Additionally fixed the clear button issue. Clicking clear now closes mega dots.
> 
> I can see there are still 2 comments you haven't responded. Could you take a look when you've got a chance? Thanks!

@yjfnyc Oh interesting. I must not have clicked the "comment" button when I answered those comments. Those two items have been addressed.

yjfnyc reviewed (approved):
let's go.
